### DISCUSSION TITLE
Global Position property for InputEventScreenTouch

### DISCUSSION THREAD

**Discussion by IDGCaptainRussia94** [godotengine/godot-proposals#9708] [Engine Core]
It would be a good addition if the event **InputEventScreenTouch** returned a global position property similar to **get_global_mouse_position**, as such I would suggest a built-in method for getting the touch position in the world, rather than having to use hacky solutions [like this](https://forum.godotengine.org/t/godot-3-0-2-get-global-position-from-touchscreen/27397/2) to get it from screen space.

**Comment by timothyqiu** [comment]
Note that `position` is not the global position. It's in viewport coordinate, i.e., global position plus canvas transform.

Also in 4.x, you can do

- `event.position * get_canvas_transform()` to get global position
- `event.position * get_global_transform_with_canvas()` to get local position

**Reply by hhyyrylainen** [reply]
I tried `event.position * get_canvas_transform()` but it doesn't seem to work. I'm trying to implement both touch and mouse inputs for a Control I have (it gets forwarded events so it doesn't know what the actual target Control was). For mouse movement `global_position` is always correct, but I can't get proper coordinates for touch inputs. 

I printed some info (I'm emulating touch from the mouse so the mouse position is what I should be getting):

```
get_canvas_transform(): [X: (1.0, 0.0), Y: (0.0, 1.0), O: (0.0, 0.0)]
Touch pos: (128.5686, 216.0685)
Mouse global position: (1197.0, 719.5)
```

And I get negative touch positions if I drag to the center of the game window. So to me it doesn't seem like the `position` actually even matches the claim in the documentation on what it should be (as I think there should be no way that there's a Viewport that is purely centered on my Control and thus any negative positions I think should be flat out impossible). 

The documentation says: `The touch position in the viewport the node is in, using the coordinate system of this viewport.`

Which to me thus just seems wrong. Instead it should say `The touch position in the coordinate system of the Control the event was received by, using the coordinate scale of its viewport.`

Is this a bug? Can I open an issue to the Godot repo or should I make a suggestion about adding `global_position` for touch events (so that I don't need to figure out how to convert the coordinates)?

**Comment by hhyyrylainen** [comment]
I opened an issue about this same thing (as this is just a discussion): https://github.com/godotengine/godot-proposals/issues/15068
